### PR TITLE
[docs] Update Adding a new module to an existing application section in Expo Modules API: Get started

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Context [Slack](https://exponent-internal.slack.com/archives/C1QMWQ1P0/p1712909879390009)

# How

- Add context to use Prebuild command to generate native project directories and use `npx expo run` commands to compile and run the example app instead of using EAS Build.
- Update Android and iOS instructions in step 3
- Update instructions in Step 2 to provide an example how to use the method from the native module

# Test Plan

Run docs locally and see: http://localhost:3002/modules/get-started/#adding-a-new-module-to-an-existing-application or see Preview link below.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/modules/get-started.mdx [hunk 1]
```diff
@@ -66,32 +75,29 @@ To run your app locally, run the `prebuild` command and then compile the app:
...
+If you have an **ios** directory in your project that is generated by `npx expo prebuild`, open the project's workspace (**ios/your-app.workspace**) in Xcode. Then, you will be able to edit the native code in **Pods** > **Development Pods** > **MyModule**.
 
-Now, change the `hello` method to return a different string. For example, you can change it to return "Hello world! 🌎🍎".
+For example, change the `hello` method to return a different string such as "Hello world! 🌎🍎". Then, run `npx expo prebuild` to generate the native directory and use `npx expo:ios` to run the example app.
 
-Rebuild the app or build a new development client and you should see your change. Remember you need to either run `npx expo prebuild` each time you make a native change or reinstall the pods using `pod install --project-directory="example/ios"` (which should be way faster).
+Remember you need to either run `npx expo prebuild` each time you make a native change or reinstall the pods using `pod install --project-directory="ios"` (which will be faster).
```
Comment (tsapeta) [inline docs/pages/modules/get-started.mdx:100]: It's not necessary to run `prebuild` after each native change. You only need to `run` it again.

`pod install` might be necessary when you modify `expo-module.config.json` (e.g. add another module class) and when you add more files to the module (they won't appear in Xcode if you add them using something else than Xcode).

**Review by tsapeta — COMMENTED** [review]
Great, thank you for improving it! I left a couple of comments.

I'm also wondering if recommending building/running the native project from Xcode and Android Studio wouldn't be better than `npx expo run` and re-running it every time you change something.

User could just run `npx expo start` once and then use the native editors which I believe offer a bit better DX for native code development. You don't need to switch between the editor and terminal, you have access to the native logs, debugger etc.

File: docs/pages/modules/get-started.mdx [hunk 2]
```diff
@@ -66,32 +75,29 @@ To run your app locally, run the `prebuild` command and then compile the app:
...
+To test and develop your module locally, use `npx expo prebuild` to generate the native project directories and use `npx expo run:android` or `npx expo run:ios` to compile and run the example app instead of using EAS Build.
 
-You can always just edit the files in the `modules/my-module/android/src/main/java/expo/modules/mymodule/` directory directly in your favorite text editor.
+#### Android
 
-Change the `hello` method to return a different string. For example, you can change it to return "Hello world! 🌎🍎".
+If you have an **android** directory in your project that is generated by `npx expo prebuild`, you can open the project directory in Android Studio. Then, you will be able to edit the native code in the **modules/my-module/android/src/main/java/expo/modules/mymodule/** directory.
 
-Rebuild the app or build a new development client and you should see your change.
+For example, change the `hello` method to return a different string such as "Hello world! 🌎🤖". Then, run `npx expo prebuild` to generate the native directory and use `npx expo:android` to run the example app.
```
Comment (amandeepmittal) [inline docs/pages/modules/get-started.mdx:92]:
```suggestion
For example, change the `hello` method to return a different string such as "Hello world! 🌎🤖". Then, run `npx expo prebuild` to generate the native directory and use `npx expo:android` to run the example app.
```

> It's not necessary to run `prebuild` after each native change. You only need to `run` it again.

@tsapeta, does the comment mean, we don't need to run `npx expo prebuild` at this step too? (similarly for iOS below).

File: docs/pages/modules/get-started.mdx [hunk 3]
```diff
@@ -25,37 +25,46 @@ Both of these flows are covered in the next sections.
...
 
 <Terminal cmd={[`$ npx create-expo-module@latest --local`]} />
 
-It's best to provide a meaningful module name. However, you can accept the default suggestions for the rest of the questions.
+You can provide a meaningful module name in the CLI prompt. For the rest of the prompts, you can also accept the default suggestions.
+
+> **Note**: If you have a pre-existing **ios** directory in your project's root which was created using `npx expo prebuild`, you must reinstall the pods:
+>
+> <br />
+> <Terminal cmd={[`$ pod install --project-directory=ios`]} />
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/modules/get-started.mdx]: Replaced here and below to use `npx pod-install`.

File: docs/pages/modules/get-started.mdx (hunk 2 above)
Comment (tsapeta) [reply docs/pages/modules/get-started.mdx]: Yes, there is no need to run `npx expo prebuild`. As we already recommend using Android Studio and Xcode, the easiest workflow in my opinion is to:
- run `npx expo prebuild` if needed (missing `ios` and `android` folders)
- open the project in AS/Xcode
- run `npx expo start` in the example app or your project (local module)
- run a build directly from AS/Xcode
- change native code
- run another build

**Comment by amandeepmittal** [comment]
@tsapeta (apologies for the delay on my end in incorporating your valuable feedback), I've updated the guide to include opening native directories in Android Studio and Xcode. Please have a look when you can.

File: docs/pages/modules/get-started.mdx [hunk 4]
```diff
@@ -11,136 +11,150 @@ import { BoxLink } from '~/ui/components/BoxLink';
...
 
-If you have an **android** directory in your project that you created using `npx expo prebuild`, you can open the directory in Android Studio.
+To develop and test your module locally, let's use Android Studio and Xcode for Android and iOS.
 
-You can always just edit the files in the `modules/my-module/android/src/main/java/expo/modules/mymodule/` directory directly in your favorite text editor.
+#### Android
 
-Change the `hello` method to return a different string. For example, you can change it to return "Hello world! 🌎🤖".
+1. Open the **android** directory (that is generated by `npx expo prebuild` in step 1) from your project in Android Studio. It may take a while for Gradle to finish syncing the native directory project.
+2. One the project is synced, open **my-module/kotlin+java/expo.modules.mymodule/MyModule** file.
```
Comment (tsapeta) [inline docs/pages/modules/get-started.mdx:87]:
```suggestion
2. Once the project is synced, open **modules/my-module/android/src/main/java/expo/modules/mymodule/MyModule.kt** file.
```

File: docs/pages/modules/get-started.mdx [hunk 5]
```diff
@@ -11,136 +11,150 @@ import { BoxLink } from '~/ui/components/BoxLink';
...
 
 <Step label="3">
 
-### Making a change
+### Edit the module
 
 #### Android
 
-Open up **MyModuleModule.kt** in Android Studio (<kbd>⌘ Cmd</kbd> + <kbd>O</kbd> or <kbd>Ctrl</kbd> + <kbd>O</kbd> and search for **MyModuleModule.kt**). Change the `hello` method to return a different string. For example, you can change it to return `"Hello world! 🌎🤖"`. Rebuild the app and you should see your change.
+1. Open **my-module/kotlin+java/expo.modules.mymodule/MyModule** file.
```
Comment (tsapeta) [inline docs/pages/modules/get-started.mdx:143]:
```suggestion
1. Open **my-module/android/src/main/java/expo/modules/mymodule/MyModule.kt** file.
```

**Review by tsapeta — APPROVED** [review]
Looks good to me, thank you!

**Review by Simek — APPROVED** [review]

**Pull request merged**